docs/reference/method/MongoDBGridFSBucket-registerGlobalStreamWrapperAlias.txt

```diff
@@ -0,0 +1,133 @@
+===========================================================
+MongoDB\\GridFS\\Bucket::registerGlobalStreamWrapperAlias()
+===========================================================
+
+.. versionadded:: 1.18
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+.. phpmethod:: MongoDB\\GridFS\\Bucket::registerGlobalStreamWrapperAlias()
+
+   Registers an alias for the bucket, which enables files within the bucket to
+   be accessed using a basic filename string (e.g.
+   `gridfs://<bucket-alias>/<filename>`).
+
+   .. code-block:: php
+
+      function registerGlobalStreamWrapperAlias(string $alias): void
+
+Parameters
+----------
+
+``$alias`` : array
+  A non-empty string used to identify the GridFS bucket when accessing files
+  using the ``gridfs://`` stream wrapper.
+
+Behavior
+--------
+
+After registering an alias for the bucket, the most recent revision of a file
+can be accessed using a filename string in the form ``gridfs://<bucket-alias>/<filename>``.
+
+Supported stream functions:
+
+- ``copy()``
+- ``file_exists()``
+- ``file_get_contents()``
+- ``file_put_contents()``
+- ``filemtime()``
+- ``filesize()``
+- ``file()``
+- ``fopen()`` (with "r", "rb", "w", and "wb" modes)
+
+In read mode, the stream context can contain the option ``gridfs['revision']``
+to specify the revision number of the file to read. If omitted, the most recent
+revision is read (revision ``-1``).
+
+In write mode, the stream context can contain the options ``gridfs['chunkSizeBytes']``
+and ``gridfs['disableMD5']`` options. If omitted, the defaults are inherited from the
```

Comment:
The `md5` field is already deprecated ([DRIVERS-2656](https://jira.mongodb.org/browse/DRIVERS-2656)). Is it worth introducing another MD5-related option into a public API at this point?

Reply:
If we do end up adding an MD5 option, let's deprecate it straight away for removal in 2.0.

Reply:
I'll remove this option as I don't expect anyone to use it. Should I set it to false by default?

Reply:
> Disabling MD5 by default.

Slight disagree if it's only done for this bucket alias API. See: https://github.com/mongodb/mongo-php-library/pull/1138#discussion_r1428158432

Reply:
So I kept the option. Just don't document it (that's actually what is in the code).